File: rxlocationmanager/src/main/java/ru/solodovnikov/rxlocationmanager/LocationTime.java
package ru.solodovnikov.rxlocationmanager;

import java.util.concurrent.TimeUnit;

public final class LocationTime {
    private final long value;
    private final TimeUnit timeUnit;

    public LocationTime(long value, TimeUnit timeUnit) {
        this.value = value;
        this.timeUnit = timeUnit;
    }

    public static LocationTime OneDay() {
        return new LocationTime(1L, TimeUnit.DAYS);
    }

    public static LocationTime OneHour() {
        return new LocationTime(1L, TimeUnit.HOURS);
    }

    long getValue() {
        return value;
    }

    TimeUnit getTimeUnit() {
        return timeUnit;
    }
}

File: core/src/main/kotlin/ru/solodovnikov/rxlocationmanager/LocationTime.kt
package ru.solodovnikov.rxlocationmanager

import java.util.concurrent.TimeUnit

/**
 * Used to describe time in the library functions
 */
data class LocationTime(val time: Long, val timeUnit: TimeUnit) {
    /**
     * Factory methods
     */
    companion object {
        /**
         * @return one day [LocationTime]
         */
        @JvmStatic
        fun oneDay() = LocationTime(1L, TimeUnit.DAYS)

        /**
         * @return one hour [LocationTime]
         */
        @JvmStatic
        fun oneHour() = LocationTime(1L, TimeUnit.HOURS)
    }
}